api/v1alpha1/vineyardruntime_types.go

```diff
@@ -0,0 +1,216 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// VineyardCompTemplateSpec is the common configurations for vineyard components including Master and Worker.
+type VineyardCompTemplateSpec struct {
+	// The replicas of Vineyard component.
+	// If not specified, defaults to 1.
+	// For worker, the replicas should not be greater than the number of nodes in the cluster
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	Replicas int32 `json:"replicas,omitempty"`
+
+	// The image of Vineyard component.
+	// For Master, the default image is `bitnami/etcd`
+	// For Worker, the default image is `vineyardcloudnative/vineyardd`
+	// The default container registry is `docker.io`, you can change it by setting the image field
+	// +optional
+	Image string `json:"image,omitempty"`
+
+	// The image tag of Vineyard component.
+	// For Master, the default image tag is `3.5.10`.
+	// For Worker, the default image tag is `latest`.
+	// +optional
+	ImageTag string `json:"imageTag,omitempty"`
+
+	// The image pull policy of Vineyard component.
+	// Default is `IfNotPresent`.
+	// +optional
+	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
+
+	// NodeSelector is a selector to choose which nodes to launch the Vineyard component.
+	// E,g. {"disktype": "ssd"}
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// Ports used by Vineyard component.
+	// For Master, the default client port is 2379 and peer port is 2380.
+	// For Worker, the default rpc port is 9600.
+	// +optional
+	Ports map[string]int `json:"ports,omitempty"`
+
+	// Environment variables that will be used by vineyard component.
+	// For Master, refer to <a href="https://etcd.io/docs/v3.5/op-guide/configuration/">Etcd Configuration</a> for more info
+	// Default is not set.
+	// +optional
+	Env map[string]string `json:"env,omitempty"`
+
+	// VolumeMounts specifies the volumes listed in ".spec.volumes" to mount into the vineyard runtime component's filesystem.
+	// It is useful for specifying a persistent storage.
+	// Default is not set.
+	// +optional
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+}
+
+// MasterSpec defines the configurations for Vineyard Master component
+// which is also regarded as the Etcd component in Vineyard.
+// For more info about Vineyard, refer to <a href="https://v6d.io/">Vineyard</a>
+type MasterSpec struct {
+	// the component configurations for Vineyard Master
+	// +optional
+	VineyardCompTemplateSpec `json:",inline"`
+
+	// whether to deploy the Vineyard Master component
+	// Default is true
+	// If set to false, the Etcd component will not be deployed.
+	// Which means the Vineyard Worker component will use an external Etcd cluster.
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+
+	// Resources contains the resource requirements and limits for the Vineyard Master.
+	// Default is not set.
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
I think master, worker and fuse needs configuring  `Resources`.

Reply:
One consideration that does not allow the master to configure resources is that users need to consider the relationship between shared memory and resources memory. If only keep the shared memory configuration, it will be more transparent to users.

Reply:
I think the meaning of transparent is that  the users don't need to configure it by default, but they can do it when they really need it.

Reply:
Okay, it looks good to me.